src/primary_names.lua

```diff
@@ -237,29 +243,34 @@ end
 
 --- Get the address for a primary name, allowing for forward lookups (e.g. "foo.bar" -> "0x123")
 --- @param name string
---- @return WalletAddress|nil address - the address for the primary name, or nil if it does not exist
+--- @return WalletAddress|nil address -- the address for the primary name, or nil if it does not exist
 function primaryNames.getAddressForPrimaryName(name)
 	return PrimaryNames.names[name]
 end
 
 --- Get the name data for an address, allowing for reverse lookups (e.g. "0x123" -> "foo.bar")
 --- @param address string
---- @return PrimaryNameWithOwner|nil primaryNameWithOwner - the primary name with owner data, or nil if it does not exist
+--- @return PrimaryNameInfo|nil -- the primary name with owner data, or nil if it does not exist
 function primaryNames.getPrimaryNameDataWithOwnerFromAddress(address)
 	local nameData = PrimaryNames.owners[address]
+
 	if not nameData then
 		return nil
 	end
+	local name = baseNameForName(nameData.name)
```

Comment:
nit: use `baseName` to avoid confusing with nameData.name

Reply:
ah, I'll condense this. was breaking it out for debug logging